src/test/llmnode_test.cpp

```diff
@@ -2604,3 +2604,205 @@ TEST_F(GetPromptTokensStringNegative, unsupportedTypesTestBool) {
+        ovms::StatusCode::OK);
+    ASSERT_STREQ(response.c_str(), "firstsecondthird");
+}
+
+TEST_F(EmbeddingsHttpTest, inputInvalid) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": 5
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer),
+        ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR);
+}
+
+TEST_F(EmbeddingsHttpTest, inputInvalidList) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": [1, "second", 3]
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer),
+        ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR);
+}
+
+TEST_F(EmbeddingsHttpTest, notExpectedField) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": "dummyInput",
+            "notExpected": "notExpected"
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer),
+        ovms::StatusCode::OK);
+    ASSERT_STREQ(response.c_str(), "dummyInput");
+}
+
+TEST_F(EmbeddingsHttpTest, invalidModel) {
+    std::string requestBody = R"(
+        {
+            "model": "invalidModel",
+            "input": "dummyInput"
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer),
+        ovms::StatusCode::MEDIAPIPE_DEFINITION_NAME_MISSING);
+}
+
+TEST_F(EmbeddingsHttpTest, encodingFormat) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": "dummyInput",
+            "encoding_format": "FORMAT"
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer),
+        ovms::StatusCode::OK);
+    ASSERT_STREQ(response.c_str(), "dummyInputFORMAT");
+}
+
+TEST_F(EmbeddingsHttpTest, invalidEncodingFormat) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": "dummyInput",
+            "encoding_format": 10
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer),
+        ovms::StatusCode::MEDIAPIPE_EXECUTION_ERROR);
+}
+
+TEST_F(EmbeddingsHttpTest, dimensions) {
+    std::string requestBody = R"(
+        {
+            "model": "embeddings",
+            "input": "dummyInput",
+            "dimensions": 2
+        }
+    )";
+    ASSERT_EQ(
+        handler->dispatchToProcessor(endpointEmbeddings, requestBody, &response, comp, responseComponents, &writer),
```

Comment:
Those tests will need to be removed, the calculator does not respond with debug string but actual embeddings...